Rust 1.31 doesn't like this PR:

```
error[E0658]: use of unstable library feature 'duration_as_u128' (see issue #50202)
   --> sticker-utils/src/progress.rs:105:48
    |
105 |             (self.n_sentences as f32 / elapsed.as_millis() as f32) * 1000f32
    |                                                ^^^^^^^^^
                                                                                
error: aborting due to previous error
                                                                                
For more information about this error, try `rustc --explain E0658`.
error: Could not compile `sticker-utils`.
To learn more, run the command again with --verbose.
The command "ci/script.sh" exited with 101.
```

--- a/sticker-utils/src/bin/sticker-server.rs
+++ b/sticker-utils/src/bin/sticker-server.rs
@@ -13,7 +13,7 @@ use sticker::depparse::{RelativePOSEncoder, RelativePositionEncoder};
 use sticker::tensorflow::{Tagger, TaggerGraph};
 use sticker::{CategoricalEncoder, LayerEncoder, Numberer, SentVectorizer, SentenceDecoder};
 use sticker_utils::{
-    sticker_app, CborRead, Config, EncoderType, LabelerType, SentProcessor, TomlRead,
+    sticker_app, CborRead, Config, EncoderType, LabelerType, SentProcessor, TaggerSpeed, TomlRead,
 };
 
 static CONFIG: &str = "CONFIG";
@@ -195,6 +195,8 @@ where
     let reader = Reader::new(BufReader::new(&conllx_stream));
     let writer = Writer::new(BufWriter::new(&conllx_stream));
 
+    let mut speed = TaggerSpeed::new();
+
     let mut sent_proc = SentProcessor::new(
         &*tagger,
         writer,
@@ -214,6 +216,8 @@ where
             let _ = writeln!(stream, "! Error processing sentence: {}", err);
             return;
         }
+
+        speed.count_sentence()
     }
 
     eprintln!("Finished processing for {}", peer_addr);

--- a/sticker-utils/src/bin/sticker-tag.rs
+++ b/sticker-utils/src/bin/sticker-tag.rs
@@ -10,7 +10,7 @@ use sticker::depparse::{RelativePOSEncoder, RelativePositionEncoder};
 use sticker::tensorflow::{Tagger, TaggerGraph};
 use sticker::{CategoricalEncoder, LayerEncoder, Numberer, SentVectorizer, SentenceDecoder};
 use sticker_utils::{
-    sticker_app, CborRead, Config, EncoderType, LabelerType, SentProcessor, TomlRead,
+    sticker_app, CborRead, Config, EncoderType, LabelerType, SentProcessor, TaggerSpeed, TomlRead,
 };
 
 static CONFIG: &str = "CONFIG";
@@ -141,6 +141,8 @@ fn process_with_decoder<D, R, W>(
     )
     .or_exit("Cannot construct tagger", 1);
 
+    let mut speed = TaggerSpeed::new();
+
     let mut sent_proc = SentProcessor::new(
         &tagger,
         write,
@@ -153,5 +155,7 @@ fn process_with_decoder<D, R, W>(
         sent_proc
             .process(sentence)
             .or_exit("Error processing sentence", 1);
+
+        speed.count_sentence()
     }
 }

--- a/sticker-utils/src/lib.rs
+++ b/sticker-utils/src/lib.rs
@@ -7,7 +7,7 @@ pub use crate::config::{
 };
 
 mod progress;
-pub use crate::progress::ReadProgress;
+pub use crate::progress::{ReadProgress, TaggerSpeed};
 
 mod sent_proc;
 pub use crate::sent_proc::SentProcessor;

--- a/sticker-utils/src/progress.rs
+++ b/sticker-utils/src/progress.rs
@@ -1,6 +1,9 @@
 use std::io::{self, Read, Seek, SeekFrom};
+use std::time::Instant;
 
 use indicatif::{ProgressBar, ProgressStyle};
+
+const NANOS_PER_SEC: u32 = 1_000_000_000;
 
 pub struct ReadProgress<R> {
     inner: R,
@@ -58,5 +61,52 @@ where
 impl<R> Drop for ReadProgress<R> {
     fn drop(&mut self) {
         self.progress_bar.finish();
+    }
+}
+
+/// Measure the number of sentences processed per second.
+///
+/// When an instance of `TaggerSpeed` is constructed, it takes the
+/// current time. `count_sentence` should be called for each sentence
+/// that was processed. A `TaggerSpeed` instance will print the
+/// processing speed to *stderr* when the instance is dropped.
+pub struct TaggerSpeed {
+    start: Instant,
+    n_sentences: usize,
+}
+
+impl TaggerSpeed {
+    /// Construct a new instance.
+    pub fn new() -> Self {
+        TaggerSpeed {
+            start: Instant::now(),
+            n_sentences: 0,
+        }
+    }
+
+    /// Count a processed sentences.
+    pub fn count_sentence(&mut self) {
+        self.n_sentences += 1;
+    }
+}
+
+impl Default for TaggerSpeed {
+    fn default() -> Self {
+        TaggerSpeed::new()
+    }
+}
+
+impl Drop for TaggerSpeed {
+    fn drop(&mut self) {
+        let elapsed = self.start.elapsed();
+        // From nightly-only as_secs_f32.
+        let elapsed_secs =
+            (elapsed.as_secs() as f32) + (elapsed.subsec_nanos() as f32) / (NANOS_PER_SEC as f32);
+        eprintln!(
+            "Processed {} sentences in {:.1}s ({:.1} sents/s)",
+            self.n_sentences,
+            elapsed_secs,
+            self.n_sentences as f32 / elapsed_secs
+        );
     }
 }

